### PR TITLE
Issue #207: update to Checkstyle 8.29

### LINKED ISSUE #207: upgrade to checkstyle 8.29
https://checkstyle.org/releasenotes.html#Release_8.29

### PR DESCRIPTION
**Pull request by Calixte**
Fixes #207.

Checked functionality of the two new checks.
config

rightcurlyenumdef

### CODE REVIEW

File: net.sf.eclipsecs.checkstyle/metadata/com/puppycrawl/tools/checkstyle/checks/coding/checkstyle-metadata.xml [hunk 1]
```diff
@@ -398,6 +405,12 @@
             <message-key key="avoid.clone.method" />
         </rule-metadata>
 
+        <rule-metadata name="%NoEnumTrailingComma.name" internal-name="NoEnumTrailingComma" parent="TreeWalker">
+            <alternative-name internal-name="com.puppycrawl.tools.checkstyle.checks.coding.NoEnumTrailingCommaCheck" />
+            <description>%NoEnumTrailingComma.desc</description>
+            <message-key key="no.enum.trailing.comma" />
+        </rule-metadata>
+
```
Comment (romani) [inline net.sf.eclipsecs.checkstyle/metadata/com/puppycrawl/tools/checkstyle/checks/coding/checkstyle-metadata.xml]: https://checkstyle.org/releasenotes.html#Release_8.29

> FallThrough module does not support the spelling "fall-through" by default.

Please update
https://github.com/checkstyle/eclipse-cs/blob/master/net.sf.eclipsecs.checkstyle/metadata/com/puppycrawl/tools/checkstyle/checks/coding/checkstyle-metadata.xml#L87

Hint: if release notes have something in "New" section .. it is almost for sure mean that eclipse plugin need be updated.

**Comment by romani** [comment]
off topic:

@Calixte ,
Please let me know if you are interested to participate at https://github.com/checkstyle/checkstyle/wiki/Checkstyle-GSoC-2020-Project-Ideas#project-name-metadata-files-for-all-modules project as mentor, to reduce cost of plugin maintenance.

File: net.sf.eclipsecs.checkstyle/metadata/com/puppycrawl/tools/checkstyle/checks/coding/checkstyle-metadata.xml (hunk 1 above)
Comment (Calixte) [reply net.sf.eclipsecs.checkstyle/metadata/com/puppycrawl/tools/checkstyle/checks/coding/checkstyle-metadata.xml]: fallthrough

Comment (romani) [reply net.sf.eclipsecs.checkstyle/metadata/com/puppycrawl/tools/checkstyle/checks/coding/checkstyle-metadata.xml]: @Calixte , it is better for reviewer to resolve items, as ack that he is satisfied.
PR author can do comment "done" or disagrement comment.

**Comment by romani** [comment]
@Bananeweizen , please do second review.

We need to figure out how to fix Travis before merge.

**Comment by Calixte** [comment]
Travis is complaining about the docs folder that #201 was supposed to fix. Interestingly, the .gitignore file you committed does not show up in master: https://github.com/checkstyle/eclipse-cs/tree/master/net.sf.eclipsecs.doc

**Comment by Calixte** [comment]
Ha, you deleted it here when you released 8.28: https://github.com/checkstyle/eclipse-cs/commit/f067b70acc9809baeb34ebd6105817342c835774#diff-2f39213c80735828c5a3193ac245e311

**Comment by romani** [comment]
very strange, I do not see in release.sh any steps that actually can do removal of this file.
I pushed this file back to master and I rebased your branch in PR, it should pass ...

**Comment by romani** [comment]
@Calixte , do you know how to resolve master build failure on openjdk12 ?
https://travis-ci.org/checkstyle/eclipse-cs/jobs/643059628#L254

**Comment by Bananeweizen** [comment]
I've tested the changes locally and created a config with the new rules, so the PR looks good to me.

Regarding Travis: It also failed with JDK12 in build 287.3 before, but it was just not recognized. Therefore there is no new issue, I believe. See console of https://travis-ci.org/checkstyle/eclipse-cs/jobs/634934769 which has the same target platform resolution issues.

I suspect that we need to upgrade Tycho (which is the Maven extension that makes Maven understand eclipse metadata). Currently we use Tycho 1.4, but only Tycho 1.5 added Java 12 support, if I get it right from here: https://wiki.eclipse.org/Tycho/Release_Notes/1.5#ECJ_update. However, we must be careful to NOT use the latest version 1.6, since that one does NO LONGER support Java 12: https://wiki.eclipse.org/Tycho/Release_Notes/1.6#Drop_support_for_obsolete_modular_JVMs_.2810_and_12.29. I can prepare a separate PR, that just sets Tycho to version 1.5.1, and then we can eventually rebase this PR, if that improves Travis.

In light of the last link, we should eventually upgrade the travis builds also, to use 13 instead of 12, since 12 had end of life already.

**Comment by romani** [comment]
@Bananeweizen , thanks a lot for investigation, lets move this to separate issue.

@Calixte , please comment out build on opnejdk12 for now, Travis will pass.

**Comment by romani** [comment]
Travis issue is moved to https://github.com/checkstyle/eclipse-cs/pull/209 .. I am merging this PR as problem already in master, so this PR do not add any problems.

**Pull request merged**